types/convert.go

```diff
@@ -169,6 +169,10 @@ func ConvertFloatToUint(sc *stmtctx.StatementContext, fval float64, upperBound u
 	}
 
 	ubf := float64(upperBound)
+	// Because u64::MAX can not be represented precisely in iee754(64bit),
+	// so u64::MAX as f64 will make a num bigger than u64::MAX,
+	// which can not be represented by 64bit integer.
+	// So (u64::MAX as f64) as u64 is undefined behavior.
```

Comment:
// Because u64::MAX can not be represented precisely in iee754(64bit),
	// `u64::MAX as f64` will make a num bigger than u64::MAX,
	// which can not be represented by 64bit integer.
	// So (u64::MAX as f64) as u64 is undefined behavior.